src/sentry/profiles/utils.py

```diff
@@ -171,3 +176,24 @@ def parse_profile_filters(query: str) -> dict[str, str]:
         profile_filters[term.key.name] = term.value.value
 
     return profile_filters
+
+
+def apply_stack_trace_rules_to_profile(profile: Profile, rules_config: str) -> None:
+    profiling_rules = keep_profiling_rules(rules_config)
+    if profiling_rules == "":
+        return
+    enhancements = Enhancements.from_config_string(profiling_rules)
+    if "version" in profile:
+        enhancements.apply_modifications_to_frame(
+            profile["profile"]["frames"], profile["platform"], {}
+        )
+    elif profile["platform"] == "android":
+        # set the fields that Enhancements expect
+        # with the right names.
+        for method in profile["profile"]["methods"]:
+            method["function"] = method.get("name", "")
+            method["abs_path"] = method.get("source_file", "")
+            method["module"] = method.get("class_name", "")
```

Comment:
Do you want to keep the redundant fields? I would clean them up after `apply_modifications_to_frame`.

Reply:
in `vroom` we don't deserialize those fields for android profiles (see [here](https://github.com/getsentry/vroom/blob/main/internal/profile/android.go#L26-L38)) so they'll be dropped automatically and won't be stored as part of the profile anyway.

This way we avoid iterating the list of frames once again.